File: I M04 - Friendship ended.java
import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        Scanner sc=new Scanner(System.in);
        String s=sc.nextLine();
        String k="Goodbye";
        int j=0;
        for(int i=0;i<s.length();i++)
        {if(s.charAt(i)==k.charAt(j))
                j+=1;
            if(j==k.length())
                break;}
        if(j==k.length())
            System.out.print("Yes");
        else
            System.out.print("No");
    }
}

File: I M04 - Friendship ended.py
s = input()
goodbye = "Goodbye"

i, j = 0, 0

while j < len(goodbye) and i < len(s):
    if s[i] == goodbye[j]:
        j += 1
    i += 1

if j == len(goodbye):
    print("Yes")
else:
    print("No")
